packages/driver/src/cy/commands/request.ts

```diff
@@ -79,349 +83,375 @@ export default (Commands, Cypress, cy, state, config) => {
     // METHOD / URL / BODY
     // or object literal with all expanded options
     request (...args) {
-      const o: any = {}
-      const userOptions = o
+      let fxArg = ''
 
-      if (_.isObject(args[0])) {
-        _.extend(userOptions, args[0])
-      } else if (args.length === 1) {
-        o.url = args[0]
-      } else if (args.length === 2) {
-        // if our first arg is a valid
-        // HTTP method then set method + url
-        if ($utils.isValidHttpMethod(args[0])) {
-          o.method = args[0]
-          o.url = args[1]
-        } else {
-          // set url + body
-          o.url = args[0]
-          o.body = args[1]
+      if (args.length === 2) {
+        fxArg = args[1]
+        // Default to POST
+        if (hasFixturePrefix(fxArg)) {
+          args = ['POST', ...args]
         }
       } else if (args.length === 3) {
-        o.method = args[0]
-        o.url = args[1]
-        o.body = args[2]
-      }
-
-      let options = _.defaults({}, userOptions, REQUEST_DEFAULTS, {
-        log: true,
-      })
-
-      // if timeout is not supplied, use the configured default
-      if (!options.timeout) {
-        options.timeout = config('responseTimeout')
+        fxArg = args[2]
       }
 
-      options.method = options.method.toUpperCase()
+      if (hasFixturePrefix(fxArg)) {
+        const fxName = fxArg.split(':')[1]
 
-      if (options.retryOnStatusCodeFailure && !options.failOnStatusCode) {
-        $errUtils.throwErrByPath('request.status_code_flags_invalid')
-      }
-
-      // normalize followRedirects -> followRedirect
-      // because we are nice
-      if (_.has(options, 'followRedirects')) {
-        options.followRedirect = options.followRedirects
-      }
+        return cy.fixture(fxName).then((fixtureData) => {
+          args[2] = fixtureData
 
-      if (!$utils.isValidHttpMethod(options.method)) {
-        $errUtils.throwErrByPath('request.invalid_method', {
-          args: { method: o.method },
+          return requestHandler(...args)
         })
       }
 
-      if (!options.url) {
-        $errUtils.throwErrByPath('request.url_missing')
-      }
-
-      if (!_.isString(options.url)) {
-        $errUtils.throwErrByPath('request.url_wrong_type')
-      }
+      return requestHandler(...args)
```

Comment:
@Crustum7 The refactor included in this change is a great suggestion, but had made it very difficult to review your changes to this file. Can you update to maintain the current file structure so you changes specifically to the command can be reviewed?

Reply:
Hi @emilyrohrbough! Sorry about the messy diff. The problem I'm having is that the request code needs to be called in two different scenarios, normally and after fixture data has been found (in the .then callback). I've therefore taken the request code and put it in a function so I can call it from each of these two places, changing the file structure in the process. I've tried to split the commits in a way that makes the diff less problematic but it doesn't seem to work. Do you have any suggestions?

Reply:
Here is how the code looked before:
```js
Commands.addAll({
    // allow our signature to be similar to cy.intercept
    // METHOD / URL / BODY
    // or object literal with all expanded options
    request (...args) {
      // old code
    },
  })
```
Here is how it looks now:
```js
Commands.addAll({
    // allow our signature to be similar to cy.intercept
    // METHOD / URL / BODY
    // or object literal with all expanded options
    request (...args) {
      let fxArg = ''

      if (args.length === 2) {
        fxArg = args[1]
        // Default to POST
        if (hasFixturePrefix(fxArg)) {
          args = ['POST', ...args]
        }
      } else if (args.length === 3) {
        fxArg = args[2]
      }

      if (hasFixturePrefix(fxArg)) {
        const fxName = fxArg.split(':')[1]

        return cy.fixture(fxName).then((fixtureData) => {
          args[2] = fixtureData

          return requestHandler(...args)
        })
      }

      return requestHandler(...args)
    },
  })

  const requestHandler = function (...args) {
    // old code (unchanged)
  }
```